.pre-commit-config.yaml

```diff
@@ -0,0 +1,42 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+# NOTE: this pre-commit file is currently only configured to run on
+#       files in hack/postrelease. If you want to add additional
+#       paths to test, update the value for files:, which is a regex.
+#
+#       See: https://pre-commit.com/index.html#regular-expressions
+
+
+default_language_version:
+  python: python3.11
```

Comment:
I'm assuming that these pre commit checks are just for our release/post-release tests since they are the only things written in python right? Will this conflict with our go code?

Reply:
There won't be any interference, no.

`pre-commit` (if you've set up the hooks) will run a series of linters/formatters before commits (to catch various issues before you commit them). The `default_language_version` setting just tells `pre-commit` to tell those linters to assume they're linting Python 3.11 code.

On top of that, the `files:` key at the bottom of the file limits `pre-commit` to only process files matching the supplied regex (in this case, `hack/postrelease/.*`) and ignore anything else.

Developers are welcome to extend the pre-commit configuration to include other checks on other code, but for the moment doing so will likely cause significant whitespace changes that no one wants so that's not enabled.

Reply:
ahhhh sorry, missed the lines just below :stuck_out_tongue: